clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1003,6 +1003,23 @@ public void onAssignment(Set<TopicPartition> assignment) {
         sensors.updatePartitionLagAndLeadSensors(assignment);
     }
 
+    /**
+     * Clear the buffered data (completed fetches) for a set of {@link TopicPartition}
+     */
+    public void clearBufferedDataForTopicPartitions(Set<TopicPartition> topicPartitionsToClear) {
+        if (topicPartitionsToClear != null && !topicPartitionsToClear.isEmpty()) {
+            Iterator<CompletedFetch> itr = completedFetches.iterator();
+            while (itr.hasNext()) {
+                TopicPartition tp = itr.next().partition;
+                if (topicPartitionsToClear.contains(tp)) {
+                    itr.remove();
+                }
+            }
+        } else if (topicPartitionsToClear != null && topicPartitionsToClear.isEmpty()) {
+            completedFetches.clear();
```

Comment:
In addition to updating `completedFetches`, should we also update `nextInLineRecords` as appropriate?